python/lsst/pipe/tasks/assembleCoadd.py

```diff
@@ -2201,6 +2216,12 @@ def findArtifacts(self, templateCoadd, tempExpRefList, imageScalerList):
                     spans.setImage(slateIm, 1, doClip=True)
                 epochCountImage += slateIm
 
+                if self.config.doFilterMorphological:
+                    fitResult = self.findSatellites.run(warpDiffExp, slateIm)
+                    satMask = afwImage.Mask(coaddBBox)
+                    satMask.array[fitResult.mask] = 1
+                    spanSetSatellite = afwGeom.SpanSet.fromMask(satMask)
```

Comment:
I am pretty sure what you want to do here is to call the `split` method on the spanset that is returned from fromMask. fromMask returns one giant SpanSet that basically just maps exactly onto the mask. But I think what you want here is a bunch of individual SpanSets with all the rest dropped. This is basically a bunch of footprints of all the satellites that are to be ignored. @yalsayyad may know better than me about what this algorithm expects though.

Also, if possible, could your findSatellites.run just use afwMasks? Or return one maybe?

Reply:
For moving the `doFilterMorphological` block above the `doPrefilterArtifacts` block, there are a couple issues. First, the satellite code uses `slateIm`, which is created from the prefiltered artifacts. I think interpolated regions are prefiltered here (as `SUSPECT`, but maybe I am wrong), and it is important that these are removed before running `findSatellites`. 
Second, I don't think prefilterArtifacts works well for the satellite trails. I tried what I think you are suggesting when I was figuring out the implementation, but ran into problems. Often the satellite trail intersects with another large footprint, like a bright star. Then, the "bad" fraction of the total footprint ends up being less than `self.prefilterArtifactsRatio`, so the whole footprint is kept, not filtered out and the satellite trail still goes into the final image.